ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5795,7 +5796,7 @@ public synchronized IMetaStoreClient getMSC(
         }
         throw ex;
       }
-      String metaStoreUris = conf.getVar(HiveConf.ConfVars.METASTOREURIS);
```

Comment:
I don't think you should be using Metastoreconf in this class.

Reply:
Some tests only set `metastore.thrift.uris`, so `conf.getVar(HiveConf.ConfVars.METASTOREURIS)` is null for this case. 
HiveMetaStoreClient checks `metastore.thrift.uris` first, and then `hive.metastore.uris`:
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java#L212

So we'd better check the `metastore.thrift.uris` first, and then `hive.metastore.uris` at here.

It's not clear to me why we shouldn't use the `Metastoreconf` in this class, this class stays in hive-exec, and we have use it else where in the same package.

Reply:
Done